python/nav/web/crispyforms.py

```diff
@@ -68,21 +69,59 @@ class NumberField(forms.IntegerField):
 
 
 class FlatFieldset:
+    """
+    A class representing a fieldset for forms.
+    Only flat layout of children fields is supported out of the box.
+    Any nesting of fields inside this fieldset might require custom
+    class definitions akin to this one.
+
+    Attributes:
+        template (str): The path to the template used for rendering the fieldset.
+        Path is relative to the app's templates directory.
+        legend (str): The legend text for the fieldset.
+        fields (list): A list of fields to include in the fieldset.
+        css_class (str): Additional CSS classes to apply to the fieldset.
+    """
+
     template = 'custom_crispy_templates/flat_fieldset.html'
 
+    def __init__(self, legend, fields: list, css_class=''):
+        """
+        Initializes a FlatFieldset object with the specified attributes.
 
-def set_flat_fieldset(legend, fields: list, css_class=''):
-    obj = FlatFieldset()
-    obj.legend = legend
-    obj.fields = fields
-    obj.css_class = css_class
-    return obj
+        Args:
+            legend (str): The legend text for the fieldset.
+            fields (list): A list of fields to include in the fieldset.
+            css_class (str, optional): Additional CSS classes to apply to the fieldset. Defaults to an empty string.
+        """
+        self.legend = legend
+        self.fields = fields
+        self.css_class = css_class
+        self.template = FlatFieldset.template
 
 
 class SubmitField:
+    """
+    A class representing a submit field (button) in a form.
+
+    Attributes:
+        name (str): The name attribute of the submit field.
+        value (str): The display text of the submit button.
+        css_classes (str): Additional CSS classes to apply to the submit button.
+        input_type (str): The type of input, which is 'submit' for this class.
+    """
+
     def __init__(
         self, name: str = 'submit', value: str = 'Submit', css_classes: str = ''
     ):
+        """
+        Initializes a SubmitField object with the specified attributes.
+
+        Args:
+            name (str, optional): The name attribute of the submit field. Defaults to 'submit'.
+            value (str, optional): The display text of the submit button. Defaults to 'Submit'.
+            css_classes (str, optional): Additional CSS classes to apply to the submit button. Defaults to an empty string.
+        """
```

Comment:
Actually, NAV uses the Sphinx convention for writing docstrings like this:

https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html

Although, with proper type annotations, I don't think you need the `:type:`  bits in the docstring any longer.